front/src/modules/timesStops/utils.ts

```diff
@@ -43,3 +47,36 @@ export const formatSuggestedViasToRowVias = (
     };
   });
 };
+
+export function findNextScheduledPoint(
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  tsPathPointsWithOpPointIndices: PathPointOpPointCorrespondance[],
+  sugOpIndex: number
+) {
+  const pathIndex = tsPathPointsWithOpPointIndices.findIndex(
+    (pathPoint) => pathPoint.correspondingOpPointIndex === sugOpIndex
+  );
+  const nextPathPoint = tsPathPointsWithOpPointIndices[pathIndex + 1];
+  return operationalPoints[nextPathPoint?.correspondingOpPointIndex];
+}
+
+const getDigits = (u: string | undefined) => (u === 's' || u === '%' ? 0 : 1);
+
+export function formatDigitsAndUnit(value: string | undefined, unit?: string) {
```

Comment:
Actually, since this function is only used three times, I think you can remove the call with theoreticalMargin in useComputedMargins and simplify the function by removing (change also the signature of the function)  
```suggestion
export function formatDigitsAndUnit(value: number | undefined, unit: string) {
```

Reply:
Same, we need a space between a value and unit.